cmake/vif.cmake

```diff
@@ -4,14 +4,31 @@
 # Generates USB-C VIF policies in XML format from device tree.
 set(gen_vif_script ${ZEPHYR_BASE}/scripts/generate_usb_vif/generate_vif.py)
 set(dts_compatible usb-c-connector)
-set(vif_xml ${PROJECT_BINARY_DIR}/vif.xml)
+set(vif_xml vif.xml)
+set(vif_out ${PROJECT_BINARY_DIR}/${vif_xml})
+
 set(cmd_gen_vif ${PYTHON_EXECUTABLE} ${gen_vif_script}
         --edt-pickle ${EDT_PICKLE}
         --compatible ${dts_compatible}
-        --vif-out ${vif_xml}
-        --board ${BOARD}
+        --vif-out ${vif_out}
         )
 
+if(CONFIG_GENVIF_INPUT_VIF_XML_PATH)
+    if (EXISTS ${APPLICATION_SOURCE_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+        set(vif_source_xml ${APPLICATION_SOURCE_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+    endif()
+else()
+    if (EXISTS ${BOARD_DIR}/${vif_xml})
+        set(vif_source_xml ${BOARD_DIR}/${vif_xml})
+    elseif (EXISTS ${APPLICATION_SOURCE_DIR}/${vif_xml})
+        set(vif_source_xml ${APPLICATION_SOURCE_DIR}/${vif_xml})
```

Comment:
Back to the original comment from Jamie, should be really be `<application>/vif.xml`? That is: does it make sense for a vif.xml to be shared with multiple boards? It may make more sense if it would lookup in `<application>/boards/<board>_vif.xml` or something like that.

Look for how other samples use the `boards` directory.